File: bookstore/src/main/java/com/luanvv/microservices/bookstore/config/UpperPhysicalNamingStrategy.java
package com.luanvv.microservices.bookstore.config;

import org.hibernate.boot.model.naming.Identifier;
import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy;

import java.util.Locale;

public class UpperPhysicalNamingStrategy extends SpringPhysicalNamingStrategy {

    @Override
    protected Identifier getIdentifier(String name, boolean quoted, JdbcEnvironment jdbcEnvironment) {
        if (isCaseInsensitive(jdbcEnvironment)) {
            name = name.toUpperCase(Locale.ROOT);
        }
        return new Identifier(name, quoted);
    }
}

File: product-import/src/main/kotlin/com/luanvv/microservices/bookstore/product_import/configuration/UpperPhysicalNamingStrategy.kt
package com.luanvv.microservices.bookstore.product_import.configuration

import org.hibernate.boot.model.naming.CamelCaseToUnderscoresNamingStrategy
import org.hibernate.boot.model.naming.Identifier
import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment
import org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy

class UpperPhysicalNamingStrategy : CamelCaseToUnderscoresNamingStrategy() {
    override fun getIdentifier(
        name: String,
        quoted: Boolean,
        jdbcEnvironment: JdbcEnvironment
    ): Identifier {
        var name = name
        if (isCaseInsensitive(jdbcEnvironment)) {
            name = name.uppercase()
        }
        return Identifier(name, quoted)
    }
}